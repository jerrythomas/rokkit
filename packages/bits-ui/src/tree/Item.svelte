<script>
	/**
	 * @typedef {Object} Props
	 * @property {string} [class]
	 */

	/** @type {Props & { [key: string]: any }} */
	let { class: className = '', children, ...rest } = $props()
</script>

<tree-item {...rest} class={className} data-tree-item>
	{@render children?.()}
</tree-item>
