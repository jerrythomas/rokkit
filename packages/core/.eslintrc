{
	"extends": "@jerrythomas/eslint-config-svelte",
	"root": true
}
