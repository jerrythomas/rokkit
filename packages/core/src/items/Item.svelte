<script>
	import { defaultFields } from '../constants'
	import Icon from '../Icon.svelte'

	export let value
	export let fields = defaultFields

	$: isObject = typeof value == 'object'
	$: text = isObject ? value[fields.text] : value
</script>

{#if isObject}
	{#if value[fields.icon]}
		{@const iconName =
			typeof value[fields.icon] == 'object'
				? value[fields.icon][value[fields.state]]
				: value[fields.icon]}
		<Icon name={iconName} />
	{:else if value[fields.image]}
		<img
			class="h-4 w-4 object-cover"
			alt={value[fields.text]}
			src={value[fields.image]}
		/>
	{/if}
{/if}
{#if text}
	<p class="w-full flex">{text}</p>
{/if}
