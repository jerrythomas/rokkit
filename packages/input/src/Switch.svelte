<script>
	import { createEventDispatcher } from 'svelte'

	const dispatch = createEventDispatcher()

	export let label = 'Use Setting'
	export let value = false
	export let tabindex = 0

	function toggle() {
		value = !value
		dispatch('change', { value })
	}
</script>

<button
	role="switch"
	class="flex rounded-full relative"
	{tabindex}
	aria-checked={value}
	class:checked={value}
	on:click={toggle}
>
	<span class="hidden">{label}</span>
	<mark
		aria-hidden="true"
		class="rounded-full absolute"
		class:is-on={value}
		class:is-off={!value}
	/>
</button>

<style>
	.is-on {
		@apply right-1;
	}
	.is-off {
		@apply left-1;
	}
</style>
