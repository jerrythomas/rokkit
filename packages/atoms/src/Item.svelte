<script>
	import Icon from './Icon.svelte'
	import { FieldMapper } from '@rokkit/core'

	/**
	 * @type {Object} Props
	 * @property {string|Object} [value]
	 * @property {FieldMapper}   [mapping]
	 */
	let { value, mapping = new FieldMapper() } = $props()
</script>

{#if mapping.hasIcon(value)}
	<Icon name={mapping.getIcon(value)} label={mapping.getLabel(value)} />
{:else if mapping.hasImage(value)}
	<img src={mapping.getImage(value)} alt={mapping.getLabel(value)} />
{/if}
<p>{mapping.getText(value)}</p>
