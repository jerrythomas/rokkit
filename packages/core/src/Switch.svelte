<script>
	import { createEventDispatcher } from 'svelte'
	import { defaultFields } from './constants'
	import { Item } from './items'
	import { navigator } from './actions'
	import { getComponent } from './lib'

	const dispatch = createEventDispatcher()

	let className = ''
	export { className as class }
	export let items = [false, true]
	export let fields = defaultFields
	export let using = {}
	export let compact = true
	export let value
	let cursor = []

	function handleNav(event) {
		value = event.detail.node
		cursor = event.detail.path

		dispatch('change', { item: value, indices: cursor })
	}

	$: useComponent = !items.every((item) => [false, true].includes(item))
	$: fields = { ...defaultFields, ...fields }
	$: using = { default: Item, ...using }
</script>

{#if !Array.isArray(items) || items.length < 2}
	<error>Items should be an array with at least two items.</error>
{:else}
	<toggle-switch
		class="flex items-center {className}"
		class:is-off={items.length == 2 && value === items[0]}
		class:is-on={items.length == 2 && value === items[1]}
		class:compact
		tabindex="0"
		role="listbox"
		use:navigator={{
			items,
			fields,
			vertical: false,
			indices: cursor
		}}
		on:move={handleNav}
		on:select={handleNav}
	>
		{#each items as item, index (item)}
			{@const component = useComponent
				? getComponent(item, fields, using)
				: null}
			<item
				class="relative flex"
				role="option"
				aria-selected={item === value}
				data-path={index}
			>
				{#if item == value}
					<indicator class="absolute bottom-0 left-0 right-0 top-0" />
				{/if}
				{#if component}
					<svelte:component this={component} value={item} {fields} />
				{/if}
			</item>
		{/each}
	</toggle-switch>
{/if}
