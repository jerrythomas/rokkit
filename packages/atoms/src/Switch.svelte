<script>
	import { equals } from 'ramda'
	import { FieldMapper, noop } from '@rokkit/core'
	import { keyboard } from '@rokkit/actions'
	import Item from './Item.svelte'

	/**
	 * @typedef {Object} Props
	 * @property {string} [class]
	 * @property {any} value
	 * @property {Array<any>} [options]
	 * @property {FieldMapper} [mapping]
	 * @property {boolean} [compact]
	 * @property {boolean} [disabled]
	 */

	/** @type {Props} */
	let {
		class: className = '',
		value = $bindable(),
		options = [false, true],
		fields = new FieldMapper({}, { default: Item }),
		compact = false,
		disabled = false,
		onchange = noop
	} = $props()

	let cursor = $state([])

	function toggle(direction = 1) {
		let nextIndex
		const index = options.indexOf(value)
		if (index === -1) {
			nextIndex = direction === -1 ? options.length - 1 : 0
		} else {
			nextIndex = (index + direction + options.length) % options.length
		}

		value = options[nextIndex]
		onchange(value)
	}

	function handleClick(event) {
		const index = event.target.closest('[data-path]').dataset.path
		value = options[index]
		onchange(value)
	}
	const keyMapping = {
		next: ['ArrowRight', 'ArrowDown', ' ', 'Enter'],
		prev: ['ArrowLeft', 'ArrowUp']
	}
	let useComponent = $derived(!options.every((item) => [false, true].includes(item)))
</script>

{#if !Array.isArray(options) || options.length < 2}
	<rkt-error>Items should be an array with at least two items.</rkt-error>
{:else}
	<!-- svelte-ignore a11y_click_events_have_key_events -->
	<rkt-switch
		class={className}
		class:is-off={options.length === 2 && equals(value, options[0])}
		class:is-on={options.length === 2 && equals(value, options[1])}
		class:compact
		aria-label="Toggle Switch"
		aria-orientation="horizontal"
		aria-disabled={disabled}
		tabindex="0"
		role="listbox"
		use:keyboard={keyMapping}
		onnext={() => toggle()}
		onprev={() => toggle(-1)}
		onclick={handleClick}
	>
		{#each options as item, index (item)}
			{@const SvelteComponent = useComponent ? fields.getComponent(item) : null}
			<rkt-item
				class="relative"
				role="option"
				aria-selected={equals(item, value)}
				data-path={index}
			>
				{#if equals(item, value)}
					<rkt-indicator class="absolute bottom-0 left-0 right-0 top-0"></rkt-indicator>
				{/if}
				{#if SvelteComponent}
					<SvelteComponent value={item} {fields} />
				{/if}
			</rkt-item>
		{/each}
	</rkt-switch>
{/if}
