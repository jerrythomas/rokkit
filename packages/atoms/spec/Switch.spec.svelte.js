import { describe, it, expect } from 'vitest'
import { render, fireEvent } from '@testing-library/svelte'
import Switch from '../src/Switch.svelte'
import { FieldMapper } from '@rokkit/core'
import Item from '../src/Item.svelte'
import { flushSync, tick } from 'svelte'

describe('Switch', () => {
	const fields = new FieldMapper({ icon: 'text' }, { default: Item })
	const nextKeys = ['ArrowDown', 'ArrowRight', 'Enter', ' ']
	const prevKeys = ['ArrowUp', 'ArrowLeft']

	it('should render', () => {
		const props = $state({ value: null })
		const { container } = render(Switch, { props })
		const switchElement = container.querySelector('rk-switch')
		expect(switchElement).toBeTruthy()
		// expect(switchElement).toHaveAttribute('role', 'listbox')
		expect(switchElement).toMatchSnapshot()

		props.class = 'small'
		flushSync()
		expect(switchElement.classList).toContain('small')
	})

	it('should render with error for invalid options', () => {
		const { container } = render(Switch, { options: ['a'] })
		const switchElement = container.querySelector('rk-switch')
		expect(switchElement).toBeFalsy()
		expect(switchElement).toMatchSnapshot()
	})
	it('should render with options', () => {
		const { container } = render(Switch, { props: { options: ['a', 'b'], fields } })
		const switchElement = container.querySelector('rk-switch')
		expect(switchElement).toBeTruthy()
		expect(switchElement).toMatchSnapshot()
	})

	it('should render with object options', () => {
		const mapper = new FieldMapper({ icon: 'text' }, { default: Item })
		const { container } = render(Switch, {
			value: { text: 'sun' },
			options: [{ text: 'sun' }, { text: 'moon' }],
			fields: mapper
		})

		const switchElement = container.querySelector('rk-switch')
		expect(switchElement).toBeTruthy()
		expect(switchElement).toMatchSnapshot()
	})

	it('should switch options on click', async () => {
		const props = $state({
			value: null,
			options: [{ text: 'sun' }, { text: 'moon' }],
			fields,
			onchange: vi.fn()
		})
		const { container } = render(Switch, { props })
		const items = container.querySelectorAll('rk-item')
		expect(items.length).toBe(2)
		fireEvent.click(items[0])
		await tick()

		expect(container).toMatchSnapshot()
		expect(props.value).toEqual({ text: 'sun' })
		expect(props.onchange).toHaveBeenCalledWith({ text: 'sun' })

		fireEvent.click(items[1])
		await tick()
		expect(container).toMatchSnapshot()
		expect(props.value).toEqual({ text: 'moon' })
		expect(props.onchange).toHaveBeenCalledWith({ text: 'moon' })
	})

	it.each(nextKeys)('should switch options on keydown %s', async (key) => {
		const props = $state({
			value: null,
			options: [{ text: 'sun' }, { text: 'moon' }],
			fields,
			onchange: vi.fn()
		})
		const { container } = render(Switch, { props })
		const switchElement = container.querySelector('rk-switch')

		fireEvent.keyUp(switchElement, { key })
		await tick()
		expect(container).toMatchSnapshot()
		expect(props.value).toEqual({ text: 'sun' })
		expect(props.onchange).toHaveBeenCalledWith({ text: 'sun' })

		fireEvent.keyUp(switchElement, { key })
		await tick()
		expect(container).toMatchSnapshot()
		expect(props.value).toEqual({ text: 'moon' })
		expect(props.onchange).toHaveBeenCalledWith({ text: 'moon' })

		fireEvent.keyUp(switchElement, { key })
		await tick()
		expect(container).toMatchSnapshot()
		expect(props.value).toEqual({ text: 'sun' })
	})

	it.each(prevKeys)('should switch options on keydown %s', async (key) => {
		const props = $state({
			value: null,
			options: [{ text: 'sun' }, { text: 'moon' }, { text: 'earth' }],
			fields,
			onchange: vi.fn()
		})
		const { container } = render(Switch, { props })
		const switchElement = container.querySelector('rk-switch')

		fireEvent.keyUp(switchElement, { key })
		await tick()
		expect(container).toMatchSnapshot()
		expect(props.value).toEqual({ text: 'earth' })
		expect(props.onchange).toHaveBeenCalledWith({ text: 'earth' })

		fireEvent.keyUp(switchElement, { key })
		await tick()
		expect(container).toMatchSnapshot()
		expect(props.value).toEqual({ text: 'moon' })
		expect(props.onchange).toHaveBeenCalledWith({ text: 'moon' })

		fireEvent.keyUp(switchElement, { key })
		await tick()
		expect(container).toMatchSnapshot()
		expect(props.value).toEqual({ text: 'sun' })
		expect(props.onchange).toHaveBeenCalledWith({ text: 'sun' })
	})
})
