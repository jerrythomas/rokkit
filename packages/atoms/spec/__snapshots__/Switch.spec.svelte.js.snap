// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Switch > should render 1`] = `
<rkt-switch
  aria-disabled="false"
  aria-label="Toggle Switch"
  aria-orientation="horizontal"
  class=""
  role="listbox"
  tabindex="0"
>
  <rkt-item
    aria-selected="false"
    class="relative"
    data-path="0"
    role="option"
  >
    <!---->
     
    <!---->
  </rkt-item>
  <rkt-item
    aria-selected="false"
    class="relative"
    data-path="1"
    role="option"
  >
    <!---->
     
    <!---->
  </rkt-item>
  
</rkt-switch>
`;

exports[`Switch > should render with error for invalid options 1`] = `null`;

exports[`Switch > should render with object options 1`] = `
<rkt-switch
  aria-disabled="false"
  aria-label="Toggle Switch"
  aria-orientation="horizontal"
  class="is-off"
  role="listbox"
  tabindex="0"
>
  <rkt-item
    aria-selected="true"
    class="relative"
    data-path="0"
    role="option"
  >
    <rkt-indicator
      class="absolute bottom-0 left-0 right-0 top-0"
    />
    <!---->
     
    <!---->
  </rkt-item>
  <rkt-item
    aria-selected="false"
    class="relative"
    data-path="1"
    role="option"
  >
    <!---->
     
    <!---->
  </rkt-item>
  
</rkt-switch>
`;

exports[`Switch > should render with options 1`] = `
<rkt-switch
  aria-disabled="false"
  aria-label="Toggle Switch"
  aria-orientation="horizontal"
  class=""
  role="listbox"
  tabindex="0"
>
  <rkt-item
    aria-selected="false"
    class="relative"
    data-path="0"
    role="option"
  >
    <!---->
     
    <!---->
  </rkt-item>
  <rkt-item
    aria-selected="false"
    class="relative"
    data-path="1"
    role="option"
  >
    <!---->
     
    <!---->
  </rkt-item>
  
</rkt-switch>
`;

exports[`Switch > should switch options on click 1`] = `
<div>
  <!---->
  <rkt-switch
    aria-disabled="false"
    aria-label="Toggle Switch"
    aria-orientation="horizontal"
    class="is-off"
    role="listbox"
    tabindex="0"
  >
    <rkt-item
      aria-selected="true"
      class="relative"
      data-path="0"
      role="option"
    >
      <rkt-indicator
        class="absolute bottom-0 left-0 right-0 top-0"
      />
      <!---->
       
      <!---->
      <!---->
      <!---->
      
      <!---->
       
      <p>
        sun
      </p>
      
      <!---->
    </rkt-item>
    <rkt-item
      aria-selected="false"
      class="relative"
      data-path="1"
      role="option"
    >
      <!---->
       
      <!---->
      <!---->
      <!---->
      
      <!---->
       
      <p>
        moon
      </p>
      
      <!---->
    </rkt-item>
    
  </rkt-switch>
  
  
</div>
`;

exports[`Switch > should switch options on click 2`] = `
<div>
  <!---->
  <rkt-switch
    aria-disabled="false"
    aria-label="Toggle Switch"
    aria-orientation="horizontal"
    class="is-on"
    role="listbox"
    tabindex="0"
  >
    <rkt-item
      aria-selected="false"
      class="relative"
      data-path="0"
      role="option"
    >
      <!---->
       
      <!---->
      <!---->
      <!---->
      
      <!---->
       
      <p>
        sun
      </p>
      
      <!---->
    </rkt-item>
    <rkt-item
      aria-selected="true"
      class="relative"
      data-path="1"
      role="option"
    >
      <rkt-indicator
        class="absolute bottom-0 left-0 right-0 top-0"
      />
      <!---->
       
      <!---->
      <!---->
      <!---->
      
      <!---->
       
      <p>
        moon
      </p>
      
      <!---->
    </rkt-item>
    
  </rkt-switch>
  
  
</div>
`;

exports[`Switch > should switch options on keydown   1`] = `
<div>
  <!---->
  <rkt-switch
    aria-disabled="false"
    aria-label="Toggle Switch"
    aria-orientation="horizontal"
    class="is-off"
    role="listbox"
    tabindex="0"
  >
    <rkt-item
      aria-selected="true"
      class="relative"
      data-path="0"
      role="option"
    >
      <rkt-indicator
        class="absolute bottom-0 left-0 right-0 top-0"
      />
      <!---->
       
      <!---->
      <!---->
      <!---->
      
      <!---->
       
      <p>
        sun
      </p>
      
      <!---->
    </rkt-item>
    <rkt-item
      aria-selected="false"
      class="relative"
      data-path="1"
      role="option"
    >
      <!---->
       
      <!---->
      <!---->
      <!---->
      
      <!---->
       
      <p>
        moon
      </p>
      
      <!---->
    </rkt-item>
    
  </rkt-switch>
  
  
</div>
`;

exports[`Switch > should switch options on keydown   2`] = `
<div>
  <!---->
  <rkt-switch
    aria-disabled="false"
    aria-label="Toggle Switch"
    aria-orientation="horizontal"
    class="is-on"
    role="listbox"
    tabindex="0"
  >
    <rkt-item
      aria-selected="false"
      class="relative"
      data-path="0"
      role="option"
    >
      <!---->
       
      <!---->
      <!---->
      <!---->
      
      <!---->
       
      <p>
        sun
      </p>
      
      <!---->
    </rkt-item>
    <rkt-item
      aria-selected="true"
      class="relative"
      data-path="1"
      role="option"
    >
      <rkt-indicator
        class="absolute bottom-0 left-0 right-0 top-0"
      />
      <!---->
       
      <!---->
      <!---->
      <!---->
      
      <!---->
       
      <p>
        moon
      </p>
      
      <!---->
    </rkt-item>
    
  </rkt-switch>
  
  
</div>
`;

exports[`Switch > should switch options on keydown   3`] = `
<div>
  <!---->
  <rkt-switch
    aria-disabled="false"
    aria-label="Toggle Switch"
    aria-orientation="horizontal"
    class="is-off"
    role="listbox"
    tabindex="0"
  >
    <rkt-item
      aria-selected="true"
      class="relative"
      data-path="0"
      role="option"
    >
      <rkt-indicator
        class="absolute bottom-0 left-0 right-0 top-0"
      />
      <!---->
       
      <!---->
      <!---->
      <!---->
      
      <!---->
       
      <p>
        sun
      </p>
      
      <!---->
    </rkt-item>
    <rkt-item
      aria-selected="false"
      class="relative"
      data-path="1"
      role="option"
    >
      <!---->
       
      <!---->
      <!---->
      <!---->
      
      <!---->
       
      <p>
        moon
      </p>
      
      <!---->
    </rkt-item>
    
  </rkt-switch>
  
  
</div>
`;

exports[`Switch > should switch options on keydown ArrowDown 1`] = `
<div>
  <!---->
  <rkt-switch
    aria-disabled="false"
    aria-label="Toggle Switch"
    aria-orientation="horizontal"
    class="is-off"
    role="listbox"
    tabindex="0"
  >
    <rkt-item
      aria-selected="true"
      class="relative"
      data-path="0"
      role="option"
    >
      <rkt-indicator
        class="absolute bottom-0 left-0 right-0 top-0"
      />
      <!---->
       
      <!---->
      <!---->
      <!---->
      
      <!---->
       
      <p>
        sun
      </p>
      
      <!---->
    </rkt-item>
    <rkt-item
      aria-selected="false"
      class="relative"
      data-path="1"
      role="option"
    >
      <!---->
       
      <!---->
      <!---->
      <!---->
      
      <!---->
       
      <p>
        moon
      </p>
      
      <!---->
    </rkt-item>
    
  </rkt-switch>
  
  
</div>
`;

exports[`Switch > should switch options on keydown ArrowDown 2`] = `
<div>
  <!---->
  <rkt-switch
    aria-disabled="false"
    aria-label="Toggle Switch"
    aria-orientation="horizontal"
    class="is-on"
    role="listbox"
    tabindex="0"
  >
    <rkt-item
      aria-selected="false"
      class="relative"
      data-path="0"
      role="option"
    >
      <!---->
       
      <!---->
      <!---->
      <!---->
      
      <!---->
       
      <p>
        sun
      </p>
      
      <!---->
    </rkt-item>
    <rkt-item
      aria-selected="true"
      class="relative"
      data-path="1"
      role="option"
    >
      <rkt-indicator
        class="absolute bottom-0 left-0 right-0 top-0"
      />
      <!---->
       
      <!---->
      <!---->
      <!---->
      
      <!---->
       
      <p>
        moon
      </p>
      
      <!---->
    </rkt-item>
    
  </rkt-switch>
  
  
</div>
`;

exports[`Switch > should switch options on keydown ArrowDown 3`] = `
<div>
  <!---->
  <rkt-switch
    aria-disabled="false"
    aria-label="Toggle Switch"
    aria-orientation="horizontal"
    class="is-off"
    role="listbox"
    tabindex="0"
  >
    <rkt-item
      aria-selected="true"
      class="relative"
      data-path="0"
      role="option"
    >
      <rkt-indicator
        class="absolute bottom-0 left-0 right-0 top-0"
      />
      <!---->
       
      <!---->
      <!---->
      <!---->
      
      <!---->
       
      <p>
        sun
      </p>
      
      <!---->
    </rkt-item>
    <rkt-item
      aria-selected="false"
      class="relative"
      data-path="1"
      role="option"
    >
      <!---->
       
      <!---->
      <!---->
      <!---->
      
      <!---->
       
      <p>
        moon
      </p>
      
      <!---->
    </rkt-item>
    
  </rkt-switch>
  
  
</div>
`;

exports[`Switch > should switch options on keydown ArrowLeft 1`] = `
<div>
  <!---->
  <rkt-switch
    aria-disabled="false"
    aria-label="Toggle Switch"
    aria-orientation="horizontal"
    class=""
    role="listbox"
    tabindex="0"
  >
    <rkt-item
      aria-selected="false"
      class="relative"
      data-path="0"
      role="option"
    >
      <!---->
       
      <!---->
      <!---->
      <!---->
      
      <!---->
       
      <p>
        sun
      </p>
      
      <!---->
    </rkt-item>
    <rkt-item
      aria-selected="false"
      class="relative"
      data-path="1"
      role="option"
    >
      <!---->
       
      <!---->
      <!---->
      <!---->
      
      <!---->
       
      <p>
        moon
      </p>
      
      <!---->
    </rkt-item>
    <rkt-item
      aria-selected="true"
      class="relative"
      data-path="2"
      role="option"
    >
      <rkt-indicator
        class="absolute bottom-0 left-0 right-0 top-0"
      />
      <!---->
       
      <!---->
      <!---->
      <!---->
      
      <!---->
       
      <p>
        earth
      </p>
      
      <!---->
    </rkt-item>
    
  </rkt-switch>
  
  
</div>
`;

exports[`Switch > should switch options on keydown ArrowLeft 2`] = `
<div>
  <!---->
  <rkt-switch
    aria-disabled="false"
    aria-label="Toggle Switch"
    aria-orientation="horizontal"
    class=""
    role="listbox"
    tabindex="0"
  >
    <rkt-item
      aria-selected="false"
      class="relative"
      data-path="0"
      role="option"
    >
      <!---->
       
      <!---->
      <!---->
      <!---->
      
      <!---->
       
      <p>
        sun
      </p>
      
      <!---->
    </rkt-item>
    <rkt-item
      aria-selected="true"
      class="relative"
      data-path="1"
      role="option"
    >
      <rkt-indicator
        class="absolute bottom-0 left-0 right-0 top-0"
      />
      <!---->
       
      <!---->
      <!---->
      <!---->
      
      <!---->
       
      <p>
        moon
      </p>
      
      <!---->
    </rkt-item>
    <rkt-item
      aria-selected="false"
      class="relative"
      data-path="2"
      role="option"
    >
      <!---->
       
      <!---->
      <!---->
      <!---->
      
      <!---->
       
      <p>
        earth
      </p>
      
      <!---->
    </rkt-item>
    
  </rkt-switch>
  
  
</div>
`;

exports[`Switch > should switch options on keydown ArrowLeft 3`] = `
<div>
  <!---->
  <rkt-switch
    aria-disabled="false"
    aria-label="Toggle Switch"
    aria-orientation="horizontal"
    class=""
    role="listbox"
    tabindex="0"
  >
    <rkt-item
      aria-selected="true"
      class="relative"
      data-path="0"
      role="option"
    >
      <rkt-indicator
        class="absolute bottom-0 left-0 right-0 top-0"
      />
      <!---->
       
      <!---->
      <!---->
      <!---->
      
      <!---->
       
      <p>
        sun
      </p>
      
      <!---->
    </rkt-item>
    <rkt-item
      aria-selected="false"
      class="relative"
      data-path="1"
      role="option"
    >
      <!---->
       
      <!---->
      <!---->
      <!---->
      
      <!---->
       
      <p>
        moon
      </p>
      
      <!---->
    </rkt-item>
    <rkt-item
      aria-selected="false"
      class="relative"
      data-path="2"
      role="option"
    >
      <!---->
       
      <!---->
      <!---->
      <!---->
      
      <!---->
       
      <p>
        earth
      </p>
      
      <!---->
    </rkt-item>
    
  </rkt-switch>
  
  
</div>
`;

exports[`Switch > should switch options on keydown ArrowRight 1`] = `
<div>
  <!---->
  <rkt-switch
    aria-disabled="false"
    aria-label="Toggle Switch"
    aria-orientation="horizontal"
    class="is-off"
    role="listbox"
    tabindex="0"
  >
    <rkt-item
      aria-selected="true"
      class="relative"
      data-path="0"
      role="option"
    >
      <rkt-indicator
        class="absolute bottom-0 left-0 right-0 top-0"
      />
      <!---->
       
      <!---->
      <!---->
      <!---->
      
      <!---->
       
      <p>
        sun
      </p>
      
      <!---->
    </rkt-item>
    <rkt-item
      aria-selected="false"
      class="relative"
      data-path="1"
      role="option"
    >
      <!---->
       
      <!---->
      <!---->
      <!---->
      
      <!---->
       
      <p>
        moon
      </p>
      
      <!---->
    </rkt-item>
    
  </rkt-switch>
  
  
</div>
`;

exports[`Switch > should switch options on keydown ArrowRight 2`] = `
<div>
  <!---->
  <rkt-switch
    aria-disabled="false"
    aria-label="Toggle Switch"
    aria-orientation="horizontal"
    class="is-on"
    role="listbox"
    tabindex="0"
  >
    <rkt-item
      aria-selected="false"
      class="relative"
      data-path="0"
      role="option"
    >
      <!---->
       
      <!---->
      <!---->
      <!---->
      
      <!---->
       
      <p>
        sun
      </p>
      
      <!---->
    </rkt-item>
    <rkt-item
      aria-selected="true"
      class="relative"
      data-path="1"
      role="option"
    >
      <rkt-indicator
        class="absolute bottom-0 left-0 right-0 top-0"
      />
      <!---->
       
      <!---->
      <!---->
      <!---->
      
      <!---->
       
      <p>
        moon
      </p>
      
      <!---->
    </rkt-item>
    
  </rkt-switch>
  
  
</div>
`;

exports[`Switch > should switch options on keydown ArrowRight 3`] = `
<div>
  <!---->
  <rkt-switch
    aria-disabled="false"
    aria-label="Toggle Switch"
    aria-orientation="horizontal"
    class="is-off"
    role="listbox"
    tabindex="0"
  >
    <rkt-item
      aria-selected="true"
      class="relative"
      data-path="0"
      role="option"
    >
      <rkt-indicator
        class="absolute bottom-0 left-0 right-0 top-0"
      />
      <!---->
       
      <!---->
      <!---->
      <!---->
      
      <!---->
       
      <p>
        sun
      </p>
      
      <!---->
    </rkt-item>
    <rkt-item
      aria-selected="false"
      class="relative"
      data-path="1"
      role="option"
    >
      <!---->
       
      <!---->
      <!---->
      <!---->
      
      <!---->
       
      <p>
        moon
      </p>
      
      <!---->
    </rkt-item>
    
  </rkt-switch>
  
  
</div>
`;

exports[`Switch > should switch options on keydown ArrowUp 1`] = `
<div>
  <!---->
  <rkt-switch
    aria-disabled="false"
    aria-label="Toggle Switch"
    aria-orientation="horizontal"
    class=""
    role="listbox"
    tabindex="0"
  >
    <rkt-item
      aria-selected="false"
      class="relative"
      data-path="0"
      role="option"
    >
      <!---->
       
      <!---->
      <!---->
      <!---->
      
      <!---->
       
      <p>
        sun
      </p>
      
      <!---->
    </rkt-item>
    <rkt-item
      aria-selected="false"
      class="relative"
      data-path="1"
      role="option"
    >
      <!---->
       
      <!---->
      <!---->
      <!---->
      
      <!---->
       
      <p>
        moon
      </p>
      
      <!---->
    </rkt-item>
    <rkt-item
      aria-selected="true"
      class="relative"
      data-path="2"
      role="option"
    >
      <rkt-indicator
        class="absolute bottom-0 left-0 right-0 top-0"
      />
      <!---->
       
      <!---->
      <!---->
      <!---->
      
      <!---->
       
      <p>
        earth
      </p>
      
      <!---->
    </rkt-item>
    
  </rkt-switch>
  
  
</div>
`;

exports[`Switch > should switch options on keydown ArrowUp 2`] = `
<div>
  <!---->
  <rkt-switch
    aria-disabled="false"
    aria-label="Toggle Switch"
    aria-orientation="horizontal"
    class=""
    role="listbox"
    tabindex="0"
  >
    <rkt-item
      aria-selected="false"
      class="relative"
      data-path="0"
      role="option"
    >
      <!---->
       
      <!---->
      <!---->
      <!---->
      
      <!---->
       
      <p>
        sun
      </p>
      
      <!---->
    </rkt-item>
    <rkt-item
      aria-selected="true"
      class="relative"
      data-path="1"
      role="option"
    >
      <rkt-indicator
        class="absolute bottom-0 left-0 right-0 top-0"
      />
      <!---->
       
      <!---->
      <!---->
      <!---->
      
      <!---->
       
      <p>
        moon
      </p>
      
      <!---->
    </rkt-item>
    <rkt-item
      aria-selected="false"
      class="relative"
      data-path="2"
      role="option"
    >
      <!---->
       
      <!---->
      <!---->
      <!---->
      
      <!---->
       
      <p>
        earth
      </p>
      
      <!---->
    </rkt-item>
    
  </rkt-switch>
  
  
</div>
`;

exports[`Switch > should switch options on keydown ArrowUp 3`] = `
<div>
  <!---->
  <rkt-switch
    aria-disabled="false"
    aria-label="Toggle Switch"
    aria-orientation="horizontal"
    class=""
    role="listbox"
    tabindex="0"
  >
    <rkt-item
      aria-selected="true"
      class="relative"
      data-path="0"
      role="option"
    >
      <rkt-indicator
        class="absolute bottom-0 left-0 right-0 top-0"
      />
      <!---->
       
      <!---->
      <!---->
      <!---->
      
      <!---->
       
      <p>
        sun
      </p>
      
      <!---->
    </rkt-item>
    <rkt-item
      aria-selected="false"
      class="relative"
      data-path="1"
      role="option"
    >
      <!---->
       
      <!---->
      <!---->
      <!---->
      
      <!---->
       
      <p>
        moon
      </p>
      
      <!---->
    </rkt-item>
    <rkt-item
      aria-selected="false"
      class="relative"
      data-path="2"
      role="option"
    >
      <!---->
       
      <!---->
      <!---->
      <!---->
      
      <!---->
       
      <p>
        earth
      </p>
      
      <!---->
    </rkt-item>
    
  </rkt-switch>
  
  
</div>
`;

exports[`Switch > should switch options on keydown Enter 1`] = `
<div>
  <!---->
  <rkt-switch
    aria-disabled="false"
    aria-label="Toggle Switch"
    aria-orientation="horizontal"
    class="is-off"
    role="listbox"
    tabindex="0"
  >
    <rkt-item
      aria-selected="true"
      class="relative"
      data-path="0"
      role="option"
    >
      <rkt-indicator
        class="absolute bottom-0 left-0 right-0 top-0"
      />
      <!---->
       
      <!---->
      <!---->
      <!---->
      
      <!---->
       
      <p>
        sun
      </p>
      
      <!---->
    </rkt-item>
    <rkt-item
      aria-selected="false"
      class="relative"
      data-path="1"
      role="option"
    >
      <!---->
       
      <!---->
      <!---->
      <!---->
      
      <!---->
       
      <p>
        moon
      </p>
      
      <!---->
    </rkt-item>
    
  </rkt-switch>
  
  
</div>
`;

exports[`Switch > should switch options on keydown Enter 2`] = `
<div>
  <!---->
  <rkt-switch
    aria-disabled="false"
    aria-label="Toggle Switch"
    aria-orientation="horizontal"
    class="is-on"
    role="listbox"
    tabindex="0"
  >
    <rkt-item
      aria-selected="false"
      class="relative"
      data-path="0"
      role="option"
    >
      <!---->
       
      <!---->
      <!---->
      <!---->
      
      <!---->
       
      <p>
        sun
      </p>
      
      <!---->
    </rkt-item>
    <rkt-item
      aria-selected="true"
      class="relative"
      data-path="1"
      role="option"
    >
      <rkt-indicator
        class="absolute bottom-0 left-0 right-0 top-0"
      />
      <!---->
       
      <!---->
      <!---->
      <!---->
      
      <!---->
       
      <p>
        moon
      </p>
      
      <!---->
    </rkt-item>
    
  </rkt-switch>
  
  
</div>
`;

exports[`Switch > should switch options on keydown Enter 3`] = `
<div>
  <!---->
  <rkt-switch
    aria-disabled="false"
    aria-label="Toggle Switch"
    aria-orientation="horizontal"
    class="is-off"
    role="listbox"
    tabindex="0"
  >
    <rkt-item
      aria-selected="true"
      class="relative"
      data-path="0"
      role="option"
    >
      <rkt-indicator
        class="absolute bottom-0 left-0 right-0 top-0"
      />
      <!---->
       
      <!---->
      <!---->
      <!---->
      
      <!---->
       
      <p>
        sun
      </p>
      
      <!---->
    </rkt-item>
    <rkt-item
      aria-selected="false"
      class="relative"
      data-path="1"
      role="option"
    >
      <!---->
       
      <!---->
      <!---->
      <!---->
      
      <!---->
       
      <p>
        moon
      </p>
      
      <!---->
    </rkt-item>
    
  </rkt-switch>
  
  
</div>
`;
